bot/seasons/evergreen/bookmark.py

```diff
@@ -50,13 +55,40 @@ def __init__(self, bot: commands.Bot):
         except discord.Forbidden:
             error_embed = discord.Embed(
                 title=random.choice(ERROR_REPLIES),
-                description=f"{ctx.author.mention}, please enable your DMs to receive the bookmark",
+                description=f"{ctx.author.mention}, please enable your DMs to receive the bookmark.",
                 colour=Colours.soft_red
             )
             await ctx.send(embed=error_embed)
         else:
-            log.info(f"{ctx.author} bookmarked {target_message.jump_url} with title '{title}'")
+            log.info(f"{ctx.author} bookmarked {target_message.jump_url} with title '{title}'.")
             await ctx.message.add_reaction(Emojis.envelope)
+            await ctx.message.add_reaction(Emojis.pin)
+
+        embed.add_field(
+            name=f'Bookmarked from {ctx.author.name}.',
+            value=f'[Visit original message]({ctx.message.jump_url})',
+            inline=False
+        )
+
+        def check(reaction: discord.Reaction, user: discord.User) -> bool:
+            return (
+                user != self.bot.user
+                and reaction.emoji == Emojis.pin
+                and reaction.message == ctx.message
+                and user not in sent_person
+            )
+
+        sent_person = {ctx.author}  # set of id who got the message
+        while True:
+            try:
+                reaction, user = await self.bot.wait_for("reaction_add", timeout=60.0, check=check)
+                log.info(f"{user.name} bookmarked {target_message.jump_url} with title '{title}' from '{ctx.author}'.")
+                with contextlib.suppress(discord.Forbidden):
+                    await user.send(embed=embed)
+                sent_person.add(user)
+            except asyncio.TimeoutError:
+                await ctx.message.clear_reactions()
+                return

```

Comment:
Since only `wait_for()` can raise a `asyncio.TimeoutError` in this block of code, it is much better to only put that in the try block, and everything else outside to reduce the level of indentation / make it clear what to do, something like this
```py
try:
    reaction, user = await self.bot.wait_for("reaction_add", timeout=60.0, check=check)
except asyncio.TimeoutError:
    # handle timeout

log.info(f"{user.name} bookmarked {target_message.jump_url} with title '{title}' from '{ctx.author}'.")
...
```
Also, as a quality of life, in case the bot cannot send the message to the user, remove the user's reaction and send a self-deleting message mentioning them that they need to open their DM - it will be more user-friendly than silently dismiss the error and let the user wonder why they did not receive any DM, similar to what you did in line 55.

Reply:
but if the chat proceed and someone randomly comes and put the reaction
it will feel weird a random error message out of nowhere.

Reply:
The bot will only send such a message if it fails to DM, and you can send a message with `delete_after` and so it will not swarm the chat at one bit.
A message like "@ someone you tried to bookmark this bookmark, but your DM is blocked, reopen it and try again" or something similar will be clear enough in my opinion.

Reply:
done